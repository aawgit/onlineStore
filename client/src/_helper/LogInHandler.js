module.exports = {
	checkLoggedIn() {
		if (sessionStorage.getItem('user')) {
			return JSON.parse(sessionStorage.getItem('user'));
		} else return false;
	},
	signOut() {
		sessionStorage.removeItem('user');
	},
};
