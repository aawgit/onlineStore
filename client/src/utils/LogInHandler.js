export const checkLoggedIn = () => {
	if (sessionStorage.getItem('user')) {
		return JSON.parse(sessionStorage.getItem('user'));
	} else return false;
};

export const signOut = () => {
	sessionStorage.removeItem('user');
};
