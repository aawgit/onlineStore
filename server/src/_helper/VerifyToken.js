var jwt = require('jsonwebtoken');
var config = require('../config');
function verifyToken(req, res, next) {
  var token = req.headers['x-access-token'];
  if (!token)
    return res.status(403).send({ auth: false, message: 'No token provided.' });
  jwt.verify(token, config.secret, function(err, decoded) {
    if (err)
    return res.status(500).send({ auth: false, message: 'Failed to authenticate token.' });
    // if everything good, save to request for use in other routes
    req.userId = decoded.id;
    next();
  });
}
module.exports = verifyToken;