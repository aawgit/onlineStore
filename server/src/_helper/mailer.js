var nodemailer = require("nodemailer");
var config = require("../config");

var transporter = nodemailer.createTransport({
  host: config.MAIL_HOST,
  Port: config.MAIL_PORT,
  secure: true,
  auth: {
    user: config.MAIL_USER,
    pass: config.MAIL_PASSWORD
  }
});

module.exports = {
  sendMail(mailOptions) {
    transporter.sendMail(mailOptions, function(err, info) {
      if (err) {
        console.log(err);
        return err;
      }
      console.log(info);
      return info;
    });
  }
};
