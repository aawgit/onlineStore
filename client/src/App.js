import React, { Component } from 'react';
import { signOut } from './_helper/LogInHandler';
import { BrowserRouter as Router } from 'react-router-dom';
import Routes from './Routes';
import { AppContext } from './AppContext';
import NavBarComp from './components/NavBarComp';
import FooterComp from './components/FooterComp';

class App extends Component {
	constructor(props) {
		super(props);
		this.setUser = this.setUser.bind(this);
		this.removeUser = this.removeUser.bind(this);
		this.state = {
			user: {
				name: '',
			},
			removeUser: () => this.removeUser(),
			setUser: (user) => this.setUser(user),
		};
	}

	componentDidMount() {
		if (sessionStorage.getItem('user')) {
			this.setState({
				user: { name: JSON.parse(sessionStorage.getItem('user')).name },
			});
		}
	}

	setUser(name) {
		this.setState({ user: { name } });
	}

	removeUser() {
		this.setState({ user: { name: '' } });
		signOut();
	}

	render() {
		return (
			<Router>
				<AppContext.Provider value={this.state}>
					<NavBarComp />
					<div className='content'>
						<Routes />
					</div>
					<FooterComp />
				</AppContext.Provider>
			</Router>
		);
	}
}

export default App;
