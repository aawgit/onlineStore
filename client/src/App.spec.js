import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { signOut } from './_helper/LogInHandler';

jest.mock('./_helper/LogInHandler');

describe('<App />', () => {
	let wrapper;

	beforeAll(() => {
		signOut.mockImplementationOnce(() => jest.fn());
		sessionStorage.setItem('user', JSON.stringify({ name: 'name1' }));
		wrapper = shallow(<App />);
	});

	it('should render', () => {
		expect(wrapper.find('Routes')).toBeTruthy();
	});

	it('should read session', () => {
		expect(wrapper.state().user.name).toEqual('name1');
	});

	it('should set user', () => {
		wrapper.instance().setUser('name2');
		expect(wrapper.state().user.name).toEqual('name2');
	});

	it('should remove user', () => {
		wrapper.instance().removeUser();
		expect(wrapper.state().user.name).toEqual('');
		expect(signOut).toHaveBeenCalledTimes(1);
	});

	it('should have no user', () => {
		wrapper.unmount();
		sessionStorage.removeItem('user', null);
		wrapper = shallow(<App />);
		expect(wrapper.state().user.name).toEqual('');
	});

	describe('state passing', () => {
		beforeEach(() => {
			wrapper.unmount();
		});

		it('removeUser', () => {
			const mock_removeUser = jest.fn();
			App.prototype.removeUser = mock_removeUser;
			wrapper = shallow(<App />);
			expect(typeof wrapper.state().removeUser).toEqual('function');
			wrapper.state().removeUser();
			expect(mock_removeUser).toHaveBeenCalledTimes(1);
		});

		it('setUser', () => {
			const mock_setUser = jest.fn();
			App.prototype.setUser = mock_setUser;
			wrapper = shallow(<App />);
			expect(typeof wrapper.state().setUser).toEqual('function');
			wrapper.state().setUser({ user: { name: 'name1' } });
			expect(mock_setUser).toHaveBeenCalledTimes(1);
		});
	});
});
