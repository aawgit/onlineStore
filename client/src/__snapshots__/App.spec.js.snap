// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <BrowserRouter>
        <NavBarComp />
        <div
          className="content"
        >
          <Routes />
        </div>
        <FooterComp />
      </BrowserRouter>,
      "value": Object {
        "removeUser": [Function],
        "setUser": [Function],
        "user": Object {
          "name": "name1",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <NavBarComp />,
          <div
            className="content"
          >
            <Routes />
          </div>,
          <FooterComp />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Routes />,
            "className": "content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <BrowserRouter>
          <NavBarComp />
          <div
            className="content"
          >
            <Routes />
          </div>
          <FooterComp />
        </BrowserRouter>,
        "value": Object {
          "removeUser": [Function],
          "setUser": [Function],
          "user": Object {
            "name": "name1",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavBarComp />,
            <div
              className="content"
            >
              <Routes />
            </div>,
            <FooterComp />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Routes />,
              "className": "content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
